src/backend/drm/mod.rs

```diff
@@ -1,38 +1,153 @@
+//! the General Buffer Manager (gbm-)API. `gbm::Buffer`s can only be rendered to by egl. Appropriate hardware-accelerated
+//! buffers are picked automatically. gbm even may fall back to `Dumbbuffers` on unsupported devices and
+//! egl-support is then provided by Mesa's software renderer `llvmpipe`.
+//!
+//! An alternative api to `gbm` is the so called `eglstream`-api, proposed (and implemented) by nvidia.
+//! It is currenly only supported by nvidias proprietary driver, which is also not accelerated by gbm, although other
+//! manifactures could in theory implement the standard, just like nvidia could implement gbm-support in their driver.
+//! Ongoing discussions about these api's may result in another (but hopefully unified) api in the future.
+//!
+//! # Implementations
+//!
+//! Smithay provided these different functionalities as wrappers around many `Device` and `Surface` implementations.
+//!
+//! At the low-level we have:
+//!
+//! - [`AtomicDrmDevice`](atomic::AtomicDrmDevice) (and [`AtomicDrmSurface`](atomic::AtomicDrmSurface)), which implement the traits directly using the modern atomic drm-api.
+//! - [`LegacyDrmDevice`](legacy::LegacyDrmDevice) (and [`LegacyDrmSurface`](legacy::LegacyDrmSurface)), which implement the traits directly using the outdated drm-api.
+//!
+//! Both of these also implement `RawDevice` and `RawSurface`.
+//!
+//! On top of that the following wrappers add buffer allocation apis:
+//!
+//! - [`GbmDevice`](gbm::GbmDevice) (and [`GbmSurface`](gbm::Surface)), which replace the direct rendering capabilities of any underlying `RawSurface` with an egl-specific `page_flip` function.
+//! - [`EglStreamDevice`](eglstream::EglStreamDevice) (and [`EglStreamSurface`](eglstream::EglStreamSurface)), which replace the direct rendering capabilities of any underlying `RawSurface` with functionality to create EGLStreams.
+//!
+//! On top of that the [`EglDevice`](egl::EglDevice) (and [`EglSurface`](egl::EglSurface)) replace the manual buffer management
+//! with an implementation of smithays [`GLGraphicsBackend`](::backend::graphics::gl::GLGraphicsBackend) to initialize the OpenGL-API for rendering.
+//!
+//! Additionally the [`FallbackDevice`](fallback::FallbackDevice) provides an easy wrapper to automatically pick an appropriate device.
+//! E.g. it can initialize an `AtomicDrmDevice` or fallback to a `LegacyDrmDevice` with the atomic api is not supported by the graphics card.
+//! It can also check for a loaded nvidia-driver for a given device and choose between an `EglStreamDevice` or a `GbmDevice`.
+//!
+//! Any layer of this chain can be implemented and therefor extended by the user. All building blocks are exposed to add addtional
+//! buffer management solutions or rendering apis out of tree. Additionally all of these implementations can be excluded from a build
+//! by disabling their corresponding feature.
+//!
+//! The most versatile type correctly provided by smithay, that should be able to initialize almost every
```

Comment:
```suggestion
//! The most versatile type currently provided by smithay, that should be able to initialize almost every
```
I think that's what you meant?

Reply:
yep